src/LibClient.cpp

```diff
@@ -137,5 +139,13 @@ namespace libclient {
     }
 
     LibClient::LibClient(std::shared_ptr<Callback> callback) : model(std::make_shared<Model>()),
-                                                               network(callback, model) {}
+                                                               network(std::move(callback), model) {}
+
+    void LibClient::setName(const std::string &name) {
+        model->clientState.name = name;
+    }
+
+    void LibClient::setRole(const spy::network::RoleEnum &role) {
+        model->clientState.role = role;
+    }
 }
```

Comment:
Die Methoden könnten auch bool zurückliefern und den state im Network mit beachten. Der Name und die Rolle sollten ja nicht verändert werden, sobald man die HelloMessage an den Server geschickt hat.

Reply:
Schau dir das mal an.
Gibts da was besseres was man prüfen konnte?

Reply:
So geht es natürlich auch.
Meine Idee war: In Network.hpp gibt es ein NetworkState enum, dass das state Attribut des Networks füllt. Man kann die Daten dann nur setzen wenn man in einem gewissen state ist (für den state müsste man noch einen getter schreiben). Der Vorteil hiervon wäre, dass man bis zur sendHelloMessage den Namen oder die Rolle nochmal ändern könnte, nachdem er gesetzt wurde. Allerdings muss man dann die "State-Machine" vom Netzwerk um einen neuen NetworkState enum typ erweitern (aktuell wird der state nicht geändert, wenn man eine Nachricht schickt, sondern nur wenn Nachrichten empfangen werden.)

Reply:
Ja klingt sinnvoll